firebase/firebase-unity-sdk
# Tools & Scripts that build testapps

Tested locally with Unity `2017.4.37f1` and  `2019.2.8f1`. 
`firestore` testapp uses [Unity Test Framework](https://docs.unity3d.com/Packages/com.unity.test-framework@1.1/manual/index.html), which requires at least Unity 2019.2. 
Except for firestore in `2017.4.37f1` and iOS analytics testapps, the rest testapps build successfully with All platform: `iOS,Android,Win64,OSXUniversal,Linux64`.
For in-editor test, MacOS' Gatekeeper (which blocks untrusted binaries) will prevent `playmode` tests on MacOS.
- Migrated Tools that automate the test. Tools are `MenuScene`, `automated_testapp` and etc., under path`scripts/gha/integration_test` 
- Migrated Scripts that build testapps. Python scripts under path `scripts/gha`. Modified the scripts and configs for GitHub.
- Updated the Unity Project setting. 

To test locally:
1. Prerequisite:
Make sure you have Unity 2017.4.37f1 installed and activated. 
For iOS, using XCode 10 by default,
For Android, using SDK tool 25.2.*, JAVA_HOME set to JDK8
(For Unity 2019.2.8f1, we need newer JDK, and Android SDK)
2. Restore secret: `python scripts/gha/restore_secrets.py --passphrase ${passphrase}`
3. Build testapps: `python scripts/gha/build_testapps.py --t auth,database --u 2017.4.37f1 --p iOS,Android,Win64,OSXUniversal,Linux64 --force_latest_runtime`
